services/core-web/src/components/mine/Reports/MineReportInfo.tsx

```diff
@@ -0,0 +1,302 @@
+        due_date_end &&
+        received_date_start &&
+        received_date_end &&
+        received_only &&
+        requested_by &&
+        status
+      );
+    });
+  };
+
+  const renderDataFromURL = (params) => {
+    const parsedParams = queryString.parse(params);
+    const filteredReports = handleFiltering(mineReports, parsedParams);
+    setFilteredReports(filteredReports);
+    setStateParams(parsedParams);
+  };
+
+  useEffect(() => {
+    if (location) {
+      renderDataFromURL(location.search);
+    }
+  }, [location]);
+
+  const fetchReports = () => {
+    return dispatch(fetchMineReports(mineGuid, mine_reports_type)).then(() => {
+      setFilteredReports(mineReports);
+    });
+  };
+
+  const handleEditReport = (report) => {
+    return dispatch(updateMineReport(report.mine_guid, report.mine_report_guid, report))
+      .then(() => dispatch(closeModal()))
+      .then(() => dispatch(fetchReports()));
+  };
+
+  const handleAddReport = (values) => {
+    return dispatch(createMineReport(mineGuid, values))
+      .then(() => dispatch(closeModal()))
+      .then(() => dispatch(fetchReports()));
+  };
+
+  const handleRemoveReport = (report) => {
+    return dispatch(deleteMineReport(report.mine_guid, report.mine_report_guid)).then(() =>
+      dispatch(fetchReports())
+    );
+  };
+
+  const openAddReportModal = (event) => {
+    event.preventDefault();
+    dispatch(
+      openModal({
+        props: {
+          onSubmit: handleAddReport,
+          title: `Add report for ${mine.mine_name}`,
+          mineGuid: mineGuid,
+          changeModalTitle: dispatch(changeModalTitle),
+          mineReportsType: mine_reports_type,
+        },
+        content: modalConfig.ADD_REPORT,
+      })
+    );
+  };
+
+  const openEditReportModal = (event, onSubmit, report) => {
+    event.preventDefault();
+    dispatch(
+      openModal({
+        props: {
+          initialValues: {
+            ...report,
+            mine_report_submission_status:
+              report.mine_report_submissions.length > 0
+                ? report.mine_report_submissions[report.mine_report_submissions.length - 1]
+                    .mine_report_submission_status_code
+                : "NRQ",
+            mineReportsType: mine_reports_type,
+          },
+          onSubmit,
+          title: `Edit ${report.submission_year} ${report.report_name}`,
+          mineGuid: mineGuid,
+          changeModalTitle: dispatch(changeModalTitle),
+        },
+        content: modalConfig.ADD_REPORT,
+      })
+    );
+  };
+
+  const handleReportFilterSubmit = (params) => {
+    setStateParams(params);
+    history.replace(routes.MINE_REPORTS.dynamicRoute(mineGuid, reportType, params));
+  };
+
+  const handleReportFilterReset = () => {
+    setStateParams({
+      ...defaultParams,
+    });
+    history.replace(routes.MINE_REPORTS.dynamicRoute(mineGuid, reportType, defaultParams));
+  };
+
+  const renderTitle = () => {
+    switch (mine_reports_type) {
```

Comment:
I'm thinking that here, could just index into MINE_REPORTS_ENUM instead of a switch

Reply:
Good call!